cypress/integration/internal/create-view-edit-delete-licence-agreements.spec.js

```diff
@@ -0,0 +1,101 @@
+const { setUp, tearDown } = require('../../support/setup');
+
+describe('View and setup new licence agreements', () => {
+  before(() => {
+    tearDown();
+    setUp('barebones');
+  });
+
+  after(() => {
+    tearDown();
+  });
+
+  it('user logs in and sets up new License Agreement', () => {
+    cy.visit(Cypress.env('ADMIN_URI'));
+    cy.fixture('users.json').then(users => {
+      cy.get('input#email').type(users.billingAndData);
+    });
+    cy.get('#password').type(Cypress.env('DEFAULT_PASSWORD'));
+    cy.get('.govuk-button.govuk-button--start').click();
+    // assert the user is signed in
+    cy.contains('Licences, users and returns');
+    cy.get('#query').clear();
+    cy.get('#query').type('AT/CURR/MONTHLY/02').should('be.visible');
+    cy.get('.search__button').click();
+    cy.contains('Licences').should('be.visible');
+    cy.get('.govuk-table__row').contains('AT/CURR/MONTHLY/02').click();
+
+    describe('sets up the sets up new License Agreemen', () => {
+      // setting up the new agreement
+      cy.get('#tab_charge').click();
+      cy.get('#charge').contains('Set up a new agreement').click();
+      cy.get('#financialAgreementCode-3').check();
+      cy.get('form > .govuk-button').click();
+      cy.get('#isDateSignedKnown-2').check();
+      cy.get('form > .govuk-button').click();
+      cy.get('#isCustomStartDate-2').click();
+      cy.get('form > .govuk-button').click();
+    // Agreement details entered and ready to submit
+    });
+
+    describe('Asserting the agreement details before submitting the form', () => {
+      cy.get('.govuk-heading-l').contains('Check agreement details').should('be.visible');
+      cy.get('.govuk-summary-list__value').contains('Canal and Rivers Trust, unsupported source (S130U)').should('be.visible');
+      cy.get('form > .govuk-button').click();
+    });
+
+    describe('Viewing, asserting the agreement details after submitting the form', () => {
+      cy.get('#tab_summary').should('be.visible');
+      cy.get('.govuk-table')
+        .children()
+        .should('contain', 'End date')
+        .should('contain', 'Agreement')
+        .should('contain', 'Date signed')
+        .should('contain', 'Action')
+        .should('contain', 'Canal and Rivers Trust, unsupported source (S130U)')
+        .should('contain', 'Delete')
+        .should('contain', 'End');
+    });
+
+    describe('End the created agreement using invalid date', () => {
+      cy.get('a.govuk-link').eq(5).click({ force: true });
```

Comment:
What's this clicking on? Is it possible to refer to the URL or the text instead of picking the elements by the order in which they're in please?

Reply:
I added contains now, this will help to understand the link we are clicking.